src/service/extensions-impl.js

```diff
@@ -168,12 +176,19 @@ export class Extensions {
   /**
    * Waits for the previously included extension to complete
    * loading/registration.
+   * @param {!Window} win
    * @param {string} extensionId
-   * @return {!Promise<!ExtensionDef>}
+   * @param {number=} opt_timeout
+   * @return {!Promise<?ExtensionDef>}
    */
-  waitForExtension(extensionId) {
-    return this.waitFor_(this.getExtensionHolder_(
-        extensionId, /* auto */ false));
+  waitForExtension(win, extensionId, opt_timeout) {
+    // Note we can't use Services.timerFor, it's a circular ref.
```

Comment:
How?

Reply:
closure complier fails `gulp check-types`

Reply:
Checked out your code, `Services.timerFor(win)` passed fine. Can you post the error being generated?

Reply:
Doh it was a test issue (the check-types was something else I battled)  I think I fixed it by installing timer service in the test, I'm running pr-check now to make sure there were no weird side effects.